src/main/scala/markatta/futiles/Cancellable.scala

```diff
@@ -0,0 +1,45 @@
+package markatta.futiles
+
+import scala.concurrent.{ExecutionContext, ExecutionException, Future, Promise}
+import scala.util.Try
+import java.util.concurrent.{Callable, FutureTask}
+
+class Cancellable[T](executionContext: ExecutionContext, block: => T) {
```

Comment:
```suggestion
final class Cancellable[T](executionContext: ExecutionContext, block: => T) {
```

Perhaps also make the constructor private and have the `apply` be the only public API to create?

Also, not sure, but could it implement/extend future (I think that is a trait) and delegate to the inner, so that a the `future` method is not needed at all?

Reply:
I ended up changing the implementation to be a delegate